lib/util/proxy.js

```diff
@@ -9,7 +10,10 @@ module.exports = function createProxySocket(proxy, target) {
       path: target.host + ':' + target.port,
       headers: { Connection: 'Keep-Alive' },
     });
-    req.on('error', reject);
+    req.on('error', error => {
+      const connectionError = new VError(`cannot connect to proxy server: ${error}`);
+      reject(connectionError);
```

Comment:
Return an actual error here so the developer knows there is a problem using their proxy server.